spec/03-dbless_spec.lua

```diff
@@ -0,0 +1,57 @@
+local helpers = require "spec.helpers"
```

Comment:
This file is not needed, it makes it look like DB less is different from others, which it isn't...

The same test should run on DB-based config.

Reply:
The `/config` endpoint used on the test is [set to explicitly fail in db-mode](https://github.com/Kong/kong/blob/master/kong/api/routes/config.lua#L24-L29). So an integration test which uses this endpoint needs to be db-less-exclusive.

That said, there *might* be a way to make the test run in db and db-lessm, by using the blueprints abstraction to create the entities (this abstraction will either call `kong.db` or write to a temp file in order to create the entities appropiatedly).

I guess it depends on how one interprets  https://github.com/kong/kong/issues/5110.
* If you "follow the issue to the letter", then we should use the `/config` endpoint, because that's what was done in the issue.
* If you "open your mind a little bit" you can make it work with db-mode as well, but then you are not exactly reproducing the issue any more (you are likely reproducing a *similar* issue which likely stems from the same root problem).

I honestly don't know what is the best approach here.

Reply:
you can use the same test (using the exit 418 in the code), but in the 01-access_spec file, as a regular plugin config and not a db-less one.

Reply:
I'm +1 for keeping this as is as a documented regression test for a reported bug.